File: backend/TaskHandlerAPI/TaskHandler.Core/Models/TaskModel.cs
using TaskHandler.Common.Enums;

namespace TaskHandler.Core.Models
{
    public class TaskModel
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; } = string.Empty;
        public Status Status { get; set; }
        public Priority Priority { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: frontend/LexisNexisFrontend/src/app/models/task.model.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  status: number; // 0 = New, 1 = In Progress, 2 = Done
  priority: number; // 0 = Low, 1 = Medium, 2 = High
  dueDate?: Date;
  createdAt: Date;
}
